### ISSUE TITLE
Creating WebCustomElements

### ISSUE THREAD

**Issue by petterek** [aspnet/Blazor#106]
I'm not sure if this is within the scope of this project, but it would be really cool of it would be possible to create webcustomelements using it. 

That way uing a blazor "element" together with other types of frameworks, or even plain HTML pages would be easy :)

**Comment by ebekker** [comment]
It might be possible to do this, but because Blazor relies on a CLR running on WASM, any such components would maintain this dependency.

**Comment by petterek** [comment]
Yes, there would have to be some kind of mechanisem in the web page to load the dll's and asure that the CLR is loaded, but I genreally like the idea of developing "autonomouse" components that can live by them self

**Comment by grahamehorner** [comment]
you could hook a wasm function to a dom element that allows to wasm then to render whatever is needed; that output could be other elements that hook to other wasm functions as I understand that you can have more than one wasm per page; not sure if/how the blazor/clr would handle that or if it currently can?

**Comment by petterek** [comment]
Hey... I understand that argument, it's more as an "feature request", and as you say most SPA has theire own component system and I think that is kind of :( using web-custom-elements would be nice.. 
I have "solved" this by regestring the first part of the name as a "namespace" and loading stuff from that. 


e.g
```
<my-custom-element>
framework.register("my","path/to/dll");
```
